delphi/translators/for2py/rectify.py

```diff
@@ -463,31 +826,117 @@ def handle_tag_statement(self, root, parElem):
                     the end of statement
                 """
                 assert is_empty(self.derived_type_var_holder_list)
-                self.derived_type_var_holder_list.append(child.attrib["id"])
-                self.parseXMLTree(child, parElem)
+                self.derived_type_var_holder_list.append(child.attrib['id'])
+                self.parseXMLTree(child, current, current, parent, traverse)
+            elif child.tag == "goto-stmt":
+                """
```

Comment:
Docstrings (strings delimited by triple-"s) should occur only at the beginning of a program unit.  See https://www.python.org/dev/peps/pep-0257/ : "A docstring is a string literal that occurs as the first statement in a module, function, class, or method definition."

This occurs at a few other places in this code as well.  They should all be fixed.